### PR TITLE
modified data structure for parent bookings and built tabs feature

### PR DESCRIPTION
**Pull request by vannguyen141290**
After working with @ehsan2793 to determined data structure, I use dummyData based on that to complete the parents' booking lists. The API endpoint is not working for some reason, but the target data structure is decided so I went ahead and build basic functionalities so that UX engineer can start working on it.
Tabs feature: now when a parent goes to booking menu, he/she can see the list of enrolled courses. 
- tab1: All >> all enrolled courses
- tab2: Today >>courses that start today
- tab3: this week >> courses that start this week
- tab4: 1:1 session

2nd commit: received comments from everyone, so I cleaned up unnecessary codes, put parents' dummyData from parentReducer to a separate file to make sure reducer file is not too long and removed eslint disable

3rd commit: I created two js files: dateHelpers.js and timeHelper.js. The purpose is to convert date and time format into desired format. I have Loom video to explain details. I discussed with @huiminzhang0111  and decided we should put these files in common directory so that everyone can use it all their components.

Loom [https://www.loom.com/share/3cf86e683c254d8394fe5965c05ffb12](url)
date and time helpers:
Loom[https://www.loom.com/share/1f9233bf012f4dcb89b7f443e13370c6](url)

### CODE REVIEW

**Review by Brendonjhawkins — APPROVED** [review]
Great work Thi!

**Review by TraNequaFauntleroy — COMMENTED** [review]
Great job fixing the functionality of the Parent booking card and cleaning up the index.js. Also, I cannot wait to get my design hands on this!

File: src/components/pages/ParentBooking/ParentBookingContainer.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  const { bookings, isFetching, error } = props.bookings;
+  const [currentTab, SetcurrentTab] = useState(null);
+
+  useEffect(() => {
+    // when the component mounts, trigger action fetchParentBookings to get the list of bookings
+    // props.fetchParentBookings();
+    SetcurrentTab(bookings);
+  }, []);
+
+  const renderTab = key => {
```
Comment (ehsan2793) [inline src/components/pages/ParentBooking/ParentBookingContainer.js:35]: Excellent work, very clean, and I love this function more than anything. It's brilliant.

**Review by ehsan2793 — APPROVED** [review]
Good job on using dummy data. We will figure out the live sever issue soon. Overall great job

File: src/components/pages/ParentBooking/ParentBookingCard.js [hunk 2]
```diff
@@ -1,23 +1,31 @@
 import React from 'react';
 import { Card } from 'antd';
 
-const ParentBookingCard = ({ sessions }) => {
-  const { course, start_date, start_time, end_date, end_time } = sessions;
+const ParentBookingCard = ({ booking }) => {
```
Comment (ryan-hamblin) [inline src/components/pages/ParentBooking/ParentBookingCard.js:4]: If you need to use the props' values you could destructure 'booking' here between these lines. 

ex: 

``` const { subject, title, ... etc} = {...booking}; ```

File: src/components/pages/ParentBooking/ParentBookingCard.js [hunk 3]
```diff
@@ -27,9 +35,9 @@ const ParentBookingCard = ({ sessions }) => {
             />
           </div>
         </div>
-        <div className="buttonWrapper">
+        {/* <div className="buttonWrapper">
```
Comment (ryan-hamblin) [inline src/components/pages/ParentBooking/ParentBookingCard.js:38]: Please delete unused or commented-out code.

File: src/components/pages/ParentBooking/ParentBookingContainer.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
+/* eslint-disable react-hooks/exhaustive-deps */
```
Comment (ryan-hamblin) [inline src/components/pages/ParentBooking/ParentBookingContainer.js:1]: Please don't disable the linter.

File: src/components/pages/ParentBooking/ParentBookingContainer.js [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+const tabs = [{ title: 'All' }, { title: 'Today' }, { title: 'This Week' }];
+
+const ParenBookingContainer = props => {
+  const { TabPane } = Tabs;
+  const { bookings, isFetching, error } = props.bookings;
+  const [currentTab, SetcurrentTab] = useState(null);
+
+  useEffect(() => {
+    // when the component mounts, trigger action fetchParentBookings to get the list of bookings
```
Comment (ryan-hamblin) [inline src/components/pages/ParentBooking/ParentBookingContainer.js:30]: I love code comments!

File: src/components/pages/ParentBooking/ParentBookingContainer.js [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+    // props.fetchParentBookings();
```
Comment (ryan-hamblin) [inline src/components/pages/ParentBooking/ParentBookingContainer.js:31]: This is commented out code and we need to remove it.

File: src/components/pages/ParentBooking/index.js [hunk 7]
```diff
@@ -25,11 +24,12 @@ function ParentBooking() {
       </Sider>
       <Content>
         <Banner />
-        <div className="bookingBtn">
+        {/* not sure if this is still needed */}
```
Comment (ryan-hamblin) [inline src/components/pages/ParentBooking/index.js:27]: Let's remove :)

File: src/components/pages/ParentHome/index.js [hunk 8]
```diff
@@ -10,13 +9,21 @@ import './../../../styles/ParentStyles/index.less';
 function ParentHome() {
   const { Content, Sider } = Layout;
   const [collapsed, setCollapsed] = useState(false);
-  const [scheduleData, setScheduleData] = useState(dummyData);
+
+  // instead of dummy data, when the component mounts, we can fetch available courses array by using getCourses action creator
+  // useEffect(() => {
```
Comment (ryan-hamblin) [inline src/components/pages/ParentHome/index.js:14]: If we don't need it, let's delete it.

File: src/components/pages/ParentHome/index.js [hunk 9]
```diff
@@ -10,13 +9,21 @@ import './../../../styles/ParentStyles/index.less';
...
+  //   console.log(token);
+  //   props.getCourses(token);
+  // }, []);
 
   const toggleCollapsed = () => {
     setCollapsed(!collapsed);
     console.log(collapsed);
   };
 
+  console.log('calendar renders');
```
Comment (ryan-hamblin) [inline src/components/pages/ParentHome/index.js:26]: Please remove this console.log

File: src/redux/reducers/parentReducer.js [hunk 10]
```diff
@@ -187,6 +190,84 @@ export const initialState = {
       sender_id: 1,
     },
   ],
+  bookings: {
+    isFetching: false,
+    bookings: [
```
Comment (ryan-hamblin) [inline src/redux/reducers/parentReducer.js:195]: Not totally necessary but if you want to reduce the size of lines inside this reducer you could define this dummy data elsewhere and import in.

File: src/components/pages/ParentHome/index.js [hunk 11]
```diff
@@ -10,13 +9,21 @@ import './../../../styles/ParentStyles/index.less';
...
+  // useEffect(() => {
+  //   const tokenObj = JSON.parse(localStorage.getItem('okta-token-storage'));
+  //   const token = tokenObj.idToken.idToken;
+  //   console.log(token);
+  //   props.getCourses(token);
+  // }, []);
 
   const toggleCollapsed = () => {
     setCollapsed(!collapsed);
     console.log(collapsed);
```
Comment (ryan-hamblin) [inline src/components/pages/ParentHome/index.js:23]: Please remove this console.log()

**Review by ryan-hamblin — DISMISSED** [review]
Wow! You accomplished a lot here and did great work especially in the instance where you're unblocking some of your teammates. 

I'd love for you to clean up the .logs and commented out code including the disable eslint rules lines. 

Also, great work on the tabs feature, but there is no description or supporting documentation as to 'why'that feature was delivered in congruency w/ the data model cleanup.

**Review by zach-morris-txt — CHANGES_REQUESTED** [review]
Needs 'merge conflicts' resolved. I will have them cleared by late evening tonight.

**Review by zach-morris-txt — APPROVED** [review]
Excellent job! Very consistent methodology...
I've pulled from main and handled the Merge Conflicts. This branch is ready to be merged.

**Pull request merged**